operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/SSABasedGenericKubernetesResourceMatcher.java

```diff
@@ -0,0 +1,338 @@
+          // this should handle the case when the value is complex in the actual map (not just a
+          // simple value).
+          result.put(keyInActual, actualMap.get(keyInActual));
+        }
+      } else {
+        // .:{} is ignored, other should not be present
+        if (!DOT_KEY.equals(key)) {
+          throw new IllegalStateException("Key: " + key + " has no prefix: " + F_PREFIX);
+        }
+      }
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void fillResultsAndTraverseFurther(Map<String, Object> result,
+      Map<String, Object> actualMap, Map<String, Object> managedFields, ObjectMapper objectMapper,
+      String key, String keyInActual, Object managedFieldValue) throws JsonProcessingException {
+    var emptyMapValue = new HashMap<String, Object>();
+    result.put(keyInActual, emptyMapValue);
+    var actualMapValue = actualMap.get(keyInActual);
+    log.debug("key: {} actual map value: {} managedFieldValue: {}", keyInActual,
+        actualMapValue, managedFieldValue);
+
+    keepOnlyManagedFields(emptyMapValue, (Map<String, Object>) actualMapValue,
+        (Map<String, Object>) managedFields.get(key), objectMapper);
+  }
+
+  private static boolean isNestedValue(Map<?, ?> managedFieldValue) {
+    return !managedFieldValue.isEmpty();
+  }
+
+  // list entries referenced by key, or when "k:" prefix is used
+  @SuppressWarnings("unchecked")
+  private static void handleListKeyEntrySet(Map<String, Object> result,
+      Map<String, Object> actualMap,
+      ObjectMapper objectMapper, String keyInActual,
+      Set<Map.Entry<String, Object>> managedEntrySet) {
+    var valueList = new ArrayList<>();
+    result.put(keyInActual, valueList);
+    var actualValueList = (List<Map<String, Object>>) actualMap.get(keyInActual);
+
+    SortedMap<Integer, Map<String, Object>> targetValuesByIndex = new TreeMap<>();
+    Map<Integer, Map<String, Object>> managedEntryByIndex = new HashMap<>();
+
+    for (Map.Entry<String, Object> listEntry : managedEntrySet) {
+      if (DOT_KEY.equals(listEntry.getKey())) {
+        continue;
+      }
+      var actualListEntry = selectListEntryBasedOnKey(keyWithoutPrefix(listEntry.getKey()),
+          actualValueList, objectMapper);
+      targetValuesByIndex.put(actualListEntry.getKey(), actualListEntry.getValue());
+      managedEntryByIndex.put(actualListEntry.getKey(), (Map<String, Object>) listEntry.getValue());
+    }
+
+    targetValuesByIndex.forEach((key, value) -> {
+      var emptyResMapValue = new HashMap<String, Object>();
+      valueList.add(emptyResMapValue);
+      try {
+        keepOnlyManagedFields(emptyResMapValue, value, managedEntryByIndex.get(key), objectMapper);
+      } catch (JsonProcessingException ex) {
+        throw new IllegalStateException(ex);
+      }
+    });
+  }
+
+  // set values, the "v:" prefix
+  @SuppressWarnings("rawtypes")
+  private static void handleSetValues(Map<String, Object> result, Map<String, Object> actualMap,
```

Comment:
Please add documentation about what this method is doing and why.

Reply:
added docs